File: LapTrinhJava/Buoi1_PhamHoDangHuy_2001215828/Chuong1/Cau1_35/KhachSan.java

class KhachSan extends BatDongSan {
    private int soSao;

    public KhachSan(String maSo, double chieuDai, double chieuRong, int soSao) {
        super(maSo, chieuDai, chieuRong);
        this.soSao = soSao;
    }

    @Override
    public double tinhGiaBan() {
        return 100000 + soSao * 50000 + chieuRong * 5000;
    }
}

File: OOP/THUC_HANH/Buoi8/Bai1/KhachSan.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Buoi8_Interface_Bai1
{
    class KhachSan : BatDongSan, IPhi
    {
        private int soSao;

        public KhachSan(string maSo, double chieuDai, double chieuRong, int soSao) : base(maSo, chieuDai, chieuRong)
        {
            this.SoSao = soSao;
        }

        protected int SoSao { get => soSao; set => soSao = value; }

        public override double GiaBan()
        {
            return 100000 + SoSao * 50000;
        }

        public double PhiKinhDoanh()
        {
            return ChieuDai * 5000;
        }

        public override void XuatThongTin()
        {
            base.XuatThongTin();
            Console.WriteLine("Số sao: " + SoSao);
            Console.WriteLine("Phí kinh doanh: " + PhiKinhDoanh());
        }
    }
}
